app/src/main/java/com/tohami/newsapi/data/repos/NewsRepository.java

```diff
@@ -0,0 +1,63 @@
+package com.tohami.newsapi.data.repos;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+
+import com.tohami.newsapi.Constants;
+import com.tohami.newsapi.base.BaseRepository;
+import com.tohami.newsapi.data.model.DataModel;
+import com.tohami.newsapi.data.model.NewsResponse;
+import com.tohami.newsapi.data.retrofit.ApiInterface;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class NewsRepository extends BaseRepository {
+
+    final private List<Call> networkCalls ;
+
+    public NewsRepository(ApiInterface newsApi) {
+        super(newsApi);
+        networkCalls = new ArrayList<>() ;
+    }
+
+    @Override
+    public void cancelAllRequests() {
+        for (Call networkCall : networkCalls) {
+            networkCall.cancel() ;
+        }
+    }
+
+
+    public LiveData<DataModel<NewsResponse>> getNews(String source, String key) {
```

Comment:
There is no check on internet connection before consuming the api

Reply:
done